src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> last = tail;
+        final Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValid(index);
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            Node<T> node = getNode(index);
+            final Node<T> previous = node.previous;
+            final Node<T> newNode = new Node<>(previous, value, node);
+            node.previous = newNode;
+            if (previous == null) {
+                head = newNode;
+            } else {
+                previous.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexIsValid(index)) {
+            return getNode(index).item;
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
```

Comment:
you use several times in your code this construction, create a separate method `checkIndex(int index)` and use it everywhere where you use such construction